src/main/java/servlets/SpendingAnalyticsServlet.java

```diff
@@ -45,22 +46,23 @@ public SpendingAnalyticsServlet(DatastoreService datastore) {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    HashMap<String, Double> storeAnalytics = getStoreAnalytics();
+    ImmutableList<HashMap<String, Double>> analytics = getAnalytics();
 
     Gson gson = new Gson();
     response.setContentType("application/json;");
-    response.getWriter().println(gson.toJson(storeAnalytics));
+    response.getWriter().println(gson.toJson(analytics));
   }
 
-  /** Returns analytics HashMap mapping a store to the total amount spent there. */
-  private HashMap<String, Double> getStoreAnalytics() {
+  /** Returns ImmutableList with information for both category and store analytics. */
+  private ImmutableList<HashMap<String, Double>> getAnalytics() {
     Query query = new Query("Receipt");
     ImmutableSet<Entity> allReceipts = datastore.prepare(query)
                                            .asList(FetchOptions.Builder.withDefaults())
                                            .stream()
                                            .collect(ImmutableSet.toImmutableSet());
 
     SpendingAnalytics analytics = new SpendingAnalytics(allReceipts);
-    return analytics.getStoreAnalytics();
+
+    return ImmutableList.of(analytics.getStoreAnalytics(), analytics.getCategoryAnalytics());
```

Comment:
Returning a `SpendingAnalytics` object and converting that to JSON would achieve the same goal as creating an `ImmutableList` out of each hash map.